src/gpio_monitor.cpp

```diff
@@ -23,7 +27,37 @@ void GpioEventHandler::setEventHandlerForGpioPresence(
 void GpioMonitor::initHandlerForGpio(
     const std::shared_ptr<boost::asio::io_context>& i_ioContext)
 {
-    // ToDo Add implementation.
-    (void)i_ioContext;
+    if (!m_sysCfgJsonObj.contains("frus"))
+    {
+        logging::logMessage("Frus section is missing in config JSON");
+        return;
+    }
+
+    try
+    {
+        for (const auto& l_Fru : m_sysCfgJsonObj["frus"].items())
+        {
+            const auto l_fruPath = l_Fru.key();
+
+            if (jsonUtility::isPollingRequired(m_sysCfgJsonObj, l_fruPath))
+            {
+                types::GpioPollingParameters l_pollingParameters =
+                    jsonUtility::getGpioPollingParameters(m_sysCfgJsonObj,
```

Comment:
This API already calls "isPollingRequired" so nesting it under "isPollingRequired" does not make sense.